src/resourcePack.ts

```diff
@@ -354,14 +355,62 @@ const prepareBlockstatesAndModels = async (progressReporter: ProgressReporter) =
     return jsons
   }
 
+  const readCustomModelData = async (path: string, namespaceDir: string) => {
+    if (!(await existsAsync(path))) return
+    const files = await fs.promises.readdir(path)
+    const customModelData = {} as Record<string, string[]>
+    await Promise.all(files.map(async (file) => {
+      const filePath = `${path}/${file}`
+      if (file.endsWith('.json')) {
+        const contents = await fs.promises.readFile(filePath, 'utf8')
+        const name = file.replace('.json', '')
+        const parsed = JSON.parse(contents)
+        const entries: string[] = []
+        if (path.endsWith('/items')) { // 1.21.4+
+          // TODO: Support other properties too
+          if (parsed.model?.type === 'range_dispatch' && parsed.model?.property === 'custom_model_data') {
```

Comment:
Can't this be used here? https://github.com/zardoy/mc-assets/blob/main/src/consumer/itemDefinitions.ts
Use cases: https://github.com/zardoy/mc-assets/blob/main/src/consumer/itemDefinitions.test.ts